### PR TITLE
database refactors

### PR DESCRIPTION
**Pull request by casey-chow**
We noticed some issues with the db schema, namely that we cannot do a has-many relationship using an integer list as Objection doesn't support those.

https://trello.com/c/b7C6ADCm/97-database-refactor

### CODE REVIEW

**Comment by casey-chow** [comment]
Most of these changes are batch renames--I'd focus time on looking at the migrations.

File: server/migrations/20190406192726_companies_point_partners.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+/* eslint-disable @typescript-eslint/camelcase */
+export async function up(knex: Knex) {
+  await knex.schema.createTable('company_point_partner', (table) => {
+    table.increments('id').primary();
+    table
+      .integer('user_id')
+      .references('id')
+      .inTable('users');
+    table
+      .integer('company_id')
```
Comment (VinaiRachakonda) [inline server/migrations/20190406192726_companies_point_partners.ts:13]: Shouldn't this be CamelCase now?

File: server/migrations/20190406192726_companies_point_partners.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import _ from 'lodash';
+import * as Knex from 'knex';
+
+/* eslint-disable @typescript-eslint/camelcase */
+export async function up(knex: Knex) {
+  await knex.schema.createTable('company_point_partner', (table) => {
+    table.increments('id').primary();
+    table
+      .integer('user_id')
```
Comment (VinaiRachakonda) [inline server/migrations/20190406192726_companies_point_partners.ts:9]: Shouldn't this be camel case?

File: server/models/user.model.ts [hunk 3]
```diff
@@ -13,16 +13,16 @@ export default class User extends BaseModel {
 
   static get relationMappings() {
     return {
-      associated_companies: {
+      companiesHandling: {
```
Comment (VinaiRachakonda) [inline server/models/user.model.ts]: Can we rename this? Maybe managedCompanies? Or at least a comment here is valuable.

File: server/migrations/20190406192726_companies_point_partners.ts (hunk 1 above)
Comment (casey-chow) [reply server/migrations/20190406192726_companies_point_partners.ts:13]: I did this refactor first and then did the rename refactor.

File: server/migrations/20190406192726_companies_point_partners.ts (hunk 2 above)
Comment (casey-chow) [reply server/migrations/20190406192726_companies_point_partners.ts:9]: I did this refactor first and then did the rename refactor.

**Comment by casey-chow** [comment]
PTAL @VinaiRachakonda

**Review by VinaiRachakonda — APPROVED** [review]

**Pull request merged**